MiscVBAFunctions/Modules/MiscOs.bas

```diff
@@ -22,3 +22,182 @@ Public Function Path(ParamArray Paths() As Variant) As String
+            If Len(EnvironVar) Then
+                IsEnvironmentalVariable = True
+            End If
+
+            If IsEnvironmentalVariable = True Then
+                EnvironVarArr(0) = EnvironVarArr(0) & EnvironVar & EnvironVarArr(2)
+                If UBound(EnvironVarArr) > 2 Then
+                    For I = 1 To UBound(EnvironVarArr) - 2
+                        EnvironVarArr(I) = EnvironVarArr(I + 2)
+                    Next
+                End If
+                ReDim Preserve EnvironVarArr(UBound(EnvironVarArr) - 2)
+                
+            Else
+                EnvironVarArr(0) = EnvironVarArr(0) & "%" & EnvironVarArr(1)
+                For I = 1 To UBound(EnvironVarArr) - 1
+                    EnvironVarArr(I) = EnvironVarArr(I + 1)
+                Next
+                ReDim Preserve EnvironVarArr(UBound(EnvironVarArr) - 1)
+            End If
+        Loop
+        
+        ' Add the last entries that can't be EnvironVar
+        For I = 1 To UBound(EnvironVarArr)
+            EnvironVarArr(0) = EnvironVarArr(0) & "%" & EnvironVarArr(I)
+        Next
+        ExpandEnvironmentalVariables = EnvironVarArr(0)
+    End If
+    
+End Function
+
+
+Private Function ConvertToBackslashes(pth As String) As String
+    ConvertToBackslashes = Replace(pth, "/", "\")
+
+End Function
+
+
+Public Function EvalPath(pth As String, Optional WB As Workbook) As String
+    ' Convert a path to absolute path. Converts system variables to String in the Path
+    '
+    ' Args:
+    '   pth: input path
+    '   WB: Optional WorkBook.
+    '
+    ' Returns:
+    '   The absolute Path.
+    
+    If WB Is Nothing Then Set WB = ThisWorkbook
+
+    EvalPath = ExpandEnvironmentalVariables(pth)
+    EvalPath = ConvertToBackslashes(EvalPath)
+
+    If Left(EvalPath, 1) Like "[A-Za-z]" And Mid(EvalPath, 2, 1) = ":" Then
+        Exit Function
+    ElseIf Left(EvalPath, 2) = "\\" Then
+        Exit Function
+    End If
+    
+    EvalPath = fso.GetAbsolutePathName(Path(WB.Path, EvalPath))
+
+End Function
+
+
+Private Function parentDir(ByVal folder)
+    parentDir = Left$(folder, InStrRev(folder, "\") - 1)
+End Function
+
+
+Public Sub CreateFolders(ByVal strPath As String, _
+                  Optional doShell As Boolean = False)
+    ' source: https://stackoverflow.com/questions/10803834/is-there-a-way-to-create-a-folder-and-sub-folders-in-excel-vba
+    ' this code will make the folders recursively if required
+    '
+    '
+    ' Args:
+    '   strPath: The desired directory
+    '   doShell: if True, the directories will be created using WScript.shell
+
+    Dim elm As Variant
+    Dim strCheckPath As String
+    Dim I As Integer
+    Dim J As Integer
+    Dim pathTempSplit() As String
+    strPath = EvalPath(strPath)
+    Dim wsh As Object
+    Set wsh = VBA.CreateObject("WScript.Shell")
+
+    Dim pathTemp As String ' for when the relative names becomes too long:
+    pathTemp = fso.GetAbsolutePathName(strPath)
```

Comment:
This will default to relative to Documents folder, I think? Need to use our evalPath rather to set it relative to this workbook?